### ISSUE TITLE
active_merchant not correctly handling authorize.net cim errors

### ISSUE THREAD

**Issue by joevandyk** [activemerchant/active_merchant#2244]
Sometimes authorize.net responds with this XML:

```
<?xml version=\"1.0\" encoding=\"utf-8\"?>
  <ErrorResponse xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns=\"AnetApi/xml/v1/schema/AnetApiSchema.xsd\">
    <messages>
      <resultCode>Error</resultCode>
      <message><code>E00104</code><text>Server in maintenance</text></message>
      <message><code>E00001</code><text>An error occurred during processing. Please try again.</text></message>
    </messages>
  </ErrorResponse>
```

Note that there's multiple messages. This will break the code in the parse method that expects there to be exactly one message.

**Comment by bl** [comment]
@joevandyk You mentioned this breaks the code in parse. From what I can tell I can't see where this would cause any unexpected behaviour. Only the first message element that appears is used.

Is the goal to transform on multiple messages?

**Comment by amandapuff** [comment]
Just ran into this error on the following line:

https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net_cim.rb#L861

Relevant stack trace:
```
TypeError: no implicit conversion of String into Integer at lib/active_support/rescuable.rb:L#80
/usr/local/lib/ruby/gems/2.3.0/gems/activemerchant-1.58.0/lib/active_merchant/billing/gateways/authorize_net_cim.rb:861:in `[]'
/usr/local/lib/ruby/gems/2.3.0/gems/activemerchant-1.58.0/lib/active_merchant/billing/gateways/authorize_net_cim.rb:861:in `commit'
/usr/local/lib/ruby/gems/2.3.0/gems/activemerchant-1.58.0/lib/active_merchant/billing/gateways/authorize_net_cim.rb:397:in `create_customer_profile_transaction'
```

Seems like `response_params['messages']` must be returning an Array, not a Hash.

**Comment by bpollack** [comment]
Seems to me as if this was resolved in #2537, so closing, but please reopen if I’ve misunderstood.
